spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/orm/jpa/HibernateProperties.java

```diff
@@ -133,6 +134,21 @@ private String determineDdlAuto(Map<String, String> existing, Supplier<String> d
 		if (ddlAuto != null) {
 			return ddlAuto;
 		}
+		String ddlDatabaseAction = existing.get(AvailableSettings.HBM2DDL_DATABASE_ACTION);
+		if (ddlDatabaseAction != null) {
```

Comment:
## Strict Approach
I chose a strict approach that only considers valid JPA database action values ([none, create, drop, drop-and-create](https://docs.jboss.org/hibernate/orm/5.2/userguide/html_single/Hibernate_User_Guide.html#configurations-hbmddl)) - each of which has a well-known counterpart `hbm2ddl.auto` value. Any other values, including valid  HBM specific values are ignored and the defaultDdlAuto provider kicks in.

## More Relaxed Approach 
However, Hibernate seems to be verify forgiving and relaxed when it comes to these values. It seems to use the JPA/HBM properties [interchangeable](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/tool/schema/spi/SchemaManagementToolCoordinator.java#L485). It prefers the JPA properties but also allows HBM values to be set on the JPA property.

A more relaxed approach would be to allow whatever values are set on the JPA property to be used as the default ddl auto and simply handle the 2 JPA special cases: `create -> create-only` and `drop-and-create -> create`. 


Anyways, let's continue the discussion.

Reply:
### Misconception
Ahh ok, I misunderstood the issue. I think my misconception was that the ddl-auto property had to be set to something even if the db-action was set.

> Rather, I think we should review this method altogether and set a ddl-auto only if necessary, 

I interpreted this to mean set the ddl-auto property if the db-action was set to a value that was a valid ddl-auto

> When checking for a user-provided setting before configuring the default setting, I think we need to consider javax.persistence.schema-generation.database.action as well as hibernate.hbm2ddl.auto.

I interpreted "configuring the default setting" as setting the ddl-auto property. Which is true but IFF the user did not already specify a ddl-auto or db-action. If the user set it already under `spring.jpa.` then there is no need for us to do anything - it will flow down to HB8 and all is well. 

### Silver lining
I learned quite a bit about the 2 properties and how we use them and how HB8 uses them.

### Third time's a charm
Pivoting here shortly and doing the following:

1. If db-action property is set, do not set a default ddl-auto (we do not require db-auto to be set if db-action is set)
2. Adjust tests 

**Question 1)** 
I feel like it would be helpful to briefly mention this property in the docs. It's not mentioned anywhere but we do support it and looking into the HB8 code they seem to prefer that property over the legacy HB8 specific one. Thoughts?

**Question 2)**
I noticed that currently we [remove the ddl-auto](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/orm/jpa/HibernateProperties.java#L107) if the default came back as "none" - which could also be from the user setting it to "none". I understand this has no functional impact because the property default is "none". I am curious as to why? Should we do the same for the db-action property? 

My **mild** concern on removing the property when user has explicitly set it to none is this: 
* User sets ddl-auto to "none"
* We remove it prior to passing through to HB8
* User steps through HB8 code in debugger
* User says "I set ddl-auto, why is it not here?"